### PR TITLE
create classes for web drivers

### PR DESCRIPTION
**Pull request by Cxristian**
for chrome and firefox also create class strateggy for driver browser

### CODE REVIEW

File: src/main/java/core/selenium/WebDriverConfig.java [hunk 1]
```diff
@@ -1,4 +1,100 @@
...
 package core.selenium;
 
+/**
+ * WebDriverConfig class.
+ *
+ * @author Cristian Lujan
+ * @version 1.0
+ */
 public class WebDriverConfig {
+
```
Comment (titor-oopart) [inline src/main/java/core/selenium/WebDriverConfig.java]: It is necessary this line?

File: src/main/java/core/selenium/webDrivers/IDriver.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+/**
+ * IDriver class.
+ *
+ * @author Cristian Lujan
+ * @version 1.0
+ */
+@FunctionalInterface
+public interface IDriver {
+    WebDriver initDriver();
```
Comment (titor-oopart) [inline src/main/java/core/selenium/webDrivers/IDriver.java:25]: add documentation

File: src/main/java/core/selenium/WebDriverConfig.java (hunk 1 above)
Comment (Cxristian) [reply src/main/java/core/selenium/WebDriverConfig.java]: this line is depend to our checkstyle

File: src/main/java/core/selenium/webDrivers/IDriver.java (hunk 2 above)
Comment (Cxristian) [reply src/main/java/core/selenium/webDrivers/IDriver.java:25]: but this line is commented in the methods used

**Review by LimbertVargas — APPROVED** [review]
It's ok for me

File: src/main/java/core/selenium/FactoryBrowser.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.Map;
+
+/**
+ * FactoryBrowser class.
+ *
+ * @author Cristian Lujan
+ * @version 1.0
+ */
+public final class FactoryBrowser {
+
```
Comment (titor-oopart) [inline src/main/java/core/selenium/FactoryBrowser.java]: delete unnecessary space

File: src/main/java/PivotalTracker/ui/BasePage.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    /**
+     * Initialises base page.
+     */
+    public BasePage() {
+        this.driver = WebDriverManager.getInstance().getWebDriver();
+        this.wait = WebDriverManager.getInstance().getWait();
+        PageFactory.initElements(driver, this);
+    }
+
```
Comment (LimbertVargas) [inline src/main/java/PivotalTracker/ui/BasePage.java]: is necessary this space?

File: src/main/java/PivotalTracker/ui/PageTransporter.java [hunk 5]
```diff
@@ -1,4 +1,55 @@
...
+import java.net.URL;
+
+/**
+ * PageTransporter class.
+ *
+ * @author Cristian Lujan
+ * @version 1.0
+ */
+public final class PageTransporter {
+    /**
```
Comment (LimbertVargas) [inline src/main/java/PivotalTracker/ui/PageTransporter.java]: add an additional space, please

File: src/main/java/PivotalTracker/ui/BasePage.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ * @author Cristian Lujan
+ * @version 1.0
+ */
+public abstract class BasePage {
+    protected WebDriver driver;
+    protected WebDriverWait wait;
+
+    /**
+     * Initialises base page.
```
Comment (aburgosu) [inline src/main/java/PivotalTracker/ui/BasePage.java]: Fix orthography Initialises --> Initializes

File: src/main/java/core/selenium/WebDriverConfig.java [hunk 7]
```diff
@@ -1,4 +1,101 @@
...
+import core.utils.ConfigFileReader;
+
+/**
+ * WebDriverConfig class.
+ *
+ * @author Cristian Lujan
+ * @version 1.0
+ */
 public class WebDriverConfig {
+    private static final String BROWSER = "browser";
```
Comment (aburgosu) [inline src/main/java/core/selenium/WebDriverConfig.java]: Verify that its necessary the constants use.

File: src/main/java/core/selenium/FactoryBrowser.java (hunk 3 above)
Comment (Cxristian) [reply src/main/java/core/selenium/FactoryBrowser.java]: done

File: src/main/java/core/selenium/WebDriverConfig.java (hunk 7 above)
Comment (Cxristian) [reply src/main/java/core/selenium/WebDriverConfig.java]: done

File: build.gradle [hunk 8]
```diff
@@ -13,6 +13,33 @@ repositories {
...
+    // set JVM arguments for the features.test JVM(s)
+    scanForTestClasses = false
+
+    //turn off Gradle's HTML report to avoid replacing the reports generated by TestNG library:
+    reports.html.enabled = true
+
+    ignoreFailures = false
+
+    systemProperty "browser", browser
+}
```
Comment (titor-oopart) [inline build.gradle]: add empty line

File: gradle.properties [hunk 9]
```diff
@@ -0,0 +1,2 @@
+#type of browser chrome/firefox
+browser=chrome
```
Comment (titor-oopart) [inline gradle.properties]: add empty line

**Review by titor-oopart — APPROVED** [review]

**Pull request merged**